### PR TITLE
Minor code tweaks to location management

### PR DESCRIPTION
**Pull request by vimto**
Some minor code-cleanup that I sliced out of the last round of refactoring. to keep PRs manageable.

### CODE REVIEW

File: packages/PlanLimitsUI/src/pages/Limits/index.tsx [hunk 1]
```diff
@@ -25,21 +24,24 @@ export const defaultViewLocation = {
...
   zoom: 8,
 };
+export const defaultPath = `/limits/${createLocationString(
+  defaultViewLocation
+)}`;
 
 export type WaterTakeFilter = 'Surface' | 'Ground' | 'Combined';
 
 export const loader: LoaderFunction = ({ params, request }) => {
+  if (!params.location) return redirect(defaultPath);
```
Comment (vimto) [inline packages/PlanLimitsUI/src/pages/Limits/index.tsx:34]: I'm a fan of this early-return code style for reducing nested conditionals, and improving readability by reducing the context required as you scan down a function.

File: packages/PlanLimitsUI/src/pages/Limits/locationString.spec.ts [hunk 2]
```diff
@@ -69,12 +63,6 @@ describe('createLocationString', () => {
 });
 
 describe('parsePinnedLocation', () => {
-  it('should return null when null passed', () => {
-    const result = parsePinnedLocation(null);
-
-    expect(result).toEqual(null);
-  });
-
```
Comment (vimto) [inline packages/PlanLimitsUI/src/pages/Limits/locationString.spec.ts:77]: TSC tells me these are redundant after tightening up the typing of the functions being tested.

File: packages/PlanLimitsUI/src/App.tsx [hunk 3]
```diff
@@ -2,32 +2,24 @@ import {
...
-      },
       {
         path: '/limits/:location',
         loader,
         element: <Limits />,
       },
+      {
+        path: '/limits',
+        element: <Navigate replace to={defaultPath} />,
+      },
```
Comment (vimto) [inline packages/PlanLimitsUI/src/App.tsx:22]: I think this simplifies our route config and removes the redundant loader needed to wrap the redirect.

File: packages/PlanLimitsUI/src/pages/Limits/index.tsx (hunk 1 above)
Comment (smozely) [reply packages/PlanLimitsUI/src/pages/Limits/index.tsx:34]: Yes, I get sometimes torn between the guideline of "one exit path for functions" and how simply something like what you've done there makes it. 
Seems like this "early-return code style" if a good place to ignore the guideline ... actually you made me do some googling http://wiki.c2.com/?GuardClause

File: packages/PlanLimitsUI/src/App.tsx (hunk 3 above)
Comment (smozely) [reply packages/PlanLimitsUI/src/App.tsx:22]: Nice, yeah thats much better

File: packages/PlanLimitsUI/src/App.tsx [hunk 4]
```diff
@@ -2,32 +2,24 @@ import {
...
+  Navigate,
 } from 'react-router-dom';
 import Layout from './Layout';
-import Limits, { defaultViewLocation, loader } from './pages/Limits';
-import { createLocationString } from './pages/Limits/locationString';
+import Limits, { defaultPath, loader } from './pages/Limits';
 
 export const routes: RouteObject[] = [
   {
-    path: '/',
```
Comment (smozely) [inline packages/PlanLimitsUI/src/App.tsx:13]: What does this mean now for visiting the app just at `/`?

File: packages/PlanLimitsUI/src/pages/Limits/locationString.ts [hunk 5]
```diff
@@ -10,43 +10,39 @@ export type ViewLocation = {
 };
 
 export function parseLocationString(
-  locationString?: string
+  locationString: string
 ): ViewLocation | null {
-  if (!locationString) {
-    return null;
-  }
+  if (!locationString) return null;
```
Comment (smozely) [inline packages/PlanLimitsUI/src/pages/Limits/locationString.ts:15]: Is this no longer needed then?

**Review by smozely — APPROVED** [review]
Good changes,

Couple of small questions, but overall lgtm

File: packages/PlanLimitsUI/src/pages/Limits/locationString.ts (hunk 5 above)
Comment (vimto) [reply packages/PlanLimitsUI/src/pages/Limits/locationString.ts:15]: Ah yep, I'll remove.

**Pull request merged**